sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java

```diff
@@ -3104,6 +3115,104 @@ public void testStorageWriteWithMultipleAppendsPerStream() throws Exception {
         containsInAnyOrder(Iterables.toArray(rows, TableRow.class)));
   }
 
+  public static class ThrowingFakeDatasetServices extends FakeDatasetService {
+    @Override
+    public BigQueryServices.StreamAppendClient getStreamAppendClient(
+        String streamName,
+        DescriptorProtos.DescriptorProto descriptor,
+        boolean useConnectionPool,
+        AppendRowsRequest.MissingValueInterpretation missingValueInterpretation) {
+      return new BigQueryServices.StreamAppendClient() {
+        @Override
+        public ApiFuture<AppendRowsResponse> appendRows(long offset, ProtoRows rows) {
```

Comment:
This is the call made to send rows to BQ. It returns an ApiFuture that may contain an exception. Here we force it to always return the exception and expect the connector to handle this properly